prov/gni/src/gnix_msg.c

```diff
@@ -55,6 +55,61 @@
 /*******************************************************************************
  * helper functions
  ******************************************************************************/
+static void __gnix_msg_unpack_data_into_iov(uint64_t dest, size_t dest_cnt,
```

Comment:
I would be more natural to define "dest" as type "struct iovec *" here and use type cast at call site when needed. The same applies to the __gnix_msg_pack_data_from_iov function.